### PR TITLE
Use clock instead time on workload controller.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Use clock instead time on workload controller.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/core/workload_controller_test.go [hunk 1]
```diff
@@ -389,17 +392,19 @@ func TestReconcile(t *testing.T) {
 					State: kueue.CheckStateReady,
 				}).
 				Condition(metav1.Condition{
-					Type:    "Admitted",
-					Status:  "True",
-					Reason:  "Admitted",
+					Type:    kueue.WorkloadAdmitted,
```
Comment (mimowo) [inline pkg/controller/core/workload_controller_test.go:395]: It's ok to use strings in tests, no need to change it imo. It is more readable that way.

Comment (mbobrovskyi) [reply pkg/controller/core/workload_controller_test.go]: Reverted

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

**Comment by mimowo** [comment]
> /test pull-kueue-test-integration-main

Opened: https://github.com/kubernetes-sigs/kueue/issues/2393

**Pull request merged**